src/core/src/memory.rs

```diff
@@ -178,12 +178,11 @@ bitflags!(
 
 ///
 #[derive(Clone, Copy, Debug)]
-pub struct MemoryBarrier;
-
-///
-#[derive(Clone, Copy, Debug)]
-pub struct BufferBarrier;
-
-///
-#[derive(Clone, Copy, Debug)]
-pub struct ImageBarrier;
+pub enum Barrier {
+    ///
+    MemoryBarrier,
```

Comment:
I don't think we need suffices here, just `Memory`, `Image`, and `Buffer` should do